pkg/initializr/initializr.go

```diff
@@ -0,0 +1,145 @@
+package initializr
+
+import (
+	"archive/zip"
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+)
+
+type InitializrRequest struct {
+	Dependencies string
+	GroupId      string
+	ArtifactId   string
+	Path         string
+}
+
+type InitializrResponse struct {
+	ContentType string
+	Contents    []byte
+	Filename    string
+}
+
+func New(request InitializrRequest) error {
+	resp, err := generate(request)
+	if err != nil {
+		return err
+	}
+	if request.Path == "" {
+		workingDir, err := os.Getwd()
+		if err != nil {
+			return err
+		}
+		return unpack(resp.Contents, workingDir)
+	} else {
+		return unpack(resp.Contents, request.Path)
+	}
+}
+
+func generate(request InitializrRequest) (InitializrResponse, error) {
+
+	u, err := url.Parse("https://start.spring.io/starter.zip")
+	q := u.Query()
+	q.Set("dependencies", request.Dependencies)
+	q.Set("groupId", request.GroupId)
+	q.Set("artifactId", request.ArtifactId)
+	q.Set("type", "maven-project")
+	u.RawQuery = q.Encode()
+
+	log.Debug("Initializr encoded URL: ", u.String())
+
+	// default timeout is infinite...
+	var httpClient = &http.Client{
+		Timeout: time.Second * 10,
+	}
+	fmt.Println("Invoking Initializr service at https://start.spring.io")
+	resp, err := httpClient.Get(u.String())
+
+	if err != nil {
+		log.Fatal(err)
```

Comment:
should return the error, I switched over to returning error vs. log.Fatal and didn't catch every usage.